search/searcher/search_phrase.go

```diff
@@ -196,7 +258,7 @@ func findPhrasePaths(prevPos uint64, ap search.ArrayPositions, phraseTerms []str
 	cdr := phraseTerms[1:]
 
 	// empty term is treated as match (continue)
-	if car == "" {
+	if len(car) == 0 || len(car) == 1 && car[0] == "" {
```

Comment:
I googled the operator precedence spec to double check this, and it's right.

On the other hand, I indeed had to look it up, so perhaps some parens might help the drive-by reader here (unless if go fmt "helpfully" strips them out for you).